### ISSUE TITLE
Conflict between check and format with D203

### ISSUE THREAD

**Issue by cnhy0** [astral-sh/ruff#12220]
```py

class Test:
    """
    Test class docstring.
    """
```
```toml
lint.select = ["D"]
lint.ignore = ["D100", "D200", "D212", "D211"]
```

When I run `ruff check test.py --config ruff.toml` I get:

```
test.py:2:5: D203 [*] 1 blank line required before class docstring
  |
1 |   class Test:
2 |       """
  |  _____^
3 | |     Test class docstring.
4 | |     """
  | |_______^ D203
  |
  = help: Insert 1 blank line before class docstring

Found 1 error.
```

With `--fix` Ruff changes the file to:

```py
class Test:

    """
    Test class docstring.
    """
```

When I now run `ruff format test.py --config ruff.toml` it redoes the changes and removes the blank line:
```py
class Test:
    """
    Test class docstring.
    """
```

Ruff version: 0.5.1

**Comment by MichaReiser** [comment]
Thanks for reporting this. It's interesting that this only comes up now :) 

I don't think there's a way to make this rule formatter compatible. But we should warn when running `ruff format` and the rule is enabled.

**Comment by cnhy0** [comment]
So it will definitely not be possible to use D203 with the formatter?

**Comment by MichaReiser** [comment]
No, I'm sorry. The formatter is designed to be compatible with black and follows its guidelines. We may want to deviate from their style guide in the future or allow more customizability, but that's something we yet have to decide on.
